### PR TITLE
chore: java doc update to for returned data instances

### LINKED ISSUE #114: Update java docs to clarify thread safety of the client
The docs should make it clear that the client instances can be safely shared across threads. The methods that return non-threadsafe instances should be prominently documented as such on both the creator method and the class returned.

For example, BigtableDataClient#newBulkMutationBatcher() can return a Batcher instance that cannot be shared across threads. The docs for newBulkMutationBatcher() should explicitly state that the returned Batcher instance is not threadsafe, so cannot be used by more than one thread. And in addition, the Batcher interface should also explicitly state in its javadocs that the interface can only be used by a single thread.

### PR DESCRIPTION
**Pull request by rahulKQL**
Fixes #114

### CODE REVIEW

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 1]
```diff
@@ -941,6 +943,8 @@ public void bulkMutateRows(BulkMutation mutation) {
    * Reads rows for given tableId in a batch. If the row does not exist, the value will be null.
    * This operation should be called with in a single thread.
    *
+   * <p>The returned Batcher instance is not threadsafe, it can only be used from single thread.
```
Comment (kolea2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: I think you can combine this with the previous sentence.

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 2]
```diff
@@ -975,6 +979,8 @@ public void bulkMutateRows(BulkMutation mutation) {
    * Reads rows for given tableId and filter criteria in a batch. If the row does not exist, the
    * value will be null. This operation should be called with in a single thread.
    *
+   * <p>The returned Batcher instance is not threadsafe, it can only be used from single thread.
```
Comment (kolea2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: same here

**Comment by kolea2** [comment]
also - conventional commits check is failing I believe because there is not a space between `chore:java...`

**Comment by rahulKQL** [comment]
Thanks for the review, I have addressed previous comments, Please have a fresh look.

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 3]
```diff
@@ -64,6 +64,9 @@
  * an application. However, close() needs to be called on the client object to clean up resources
  * such as threads during application shutdown.
  *
+ * <p>This client can be safely shared across multiple threads except for the Batcher instances
+ * returned from bulk operations, eg. `newBulkMutationBatcher()`, `newBulkReadRowsBatcher()`.
+ *
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: Also, the returned `ServerStream` from  `readRows` can't be used across multiple threads

Comment (rahulKQL) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: Thanks for the review!!!

I have added similar comment on `BigtableDataClient#readRows()`. PTAL

**Pull request merged**